Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -211,10 +217,37 @@ private extension AppSettingsStore {
         onCompletion(.success(settings.isViewAddOnsSwitchEnabled))
     }
 
+    /// Loads the last persisted eligibility error information from `GeneralAppSettings`
+    ///
+    func loadEligibilityErrorInfo(onCompletion: (Result<EligibilityErrorInfo, Error>) -> Void) {
+        let settings = loadOrCreateGeneralAppSettings()
+
+        guard let errorInfo = settings.lastEligibilityErrorInfo else {
+            onCompletion(.failure(AppSettingsStoreErrors.noEligibilityErrorInfo))
+            return
```

Comment:
super nit: This could be a one liner!
```suggestion
            return onCompletion(.failure(AppSettingsStoreErrors.noEligibilityErrorInfo))
```

Reply:
Very cool! ❄️